src/toil/test/lib/dockerTest.py

```diff
@@ -37,66 +39,71 @@ def testDockerClean(self, caching=True):
+        options.logLevel = 'OFF'
         options.workDir = work_dir
         options.clean = 'always'
+        options.disableCaching = False
         if not caching:
             options.disableCaching = True
+
+        _logger.info("toilDev DockerTest")
+
         for rm in (True, False):
             for detached in (True, False):
                 if detached and rm:
                     continue
-                for defer in (FORGO, STOP, RM, None):
+                for deferParam in (FORGO, STOP, RM, None):
                     # Not using base64 logic here since it might create a name starting with a `-`.
                     container_name = uuid.uuid4().hex
-                    A = Job.wrapJobFn(_testDockerCleanFn, data_dir, detached, rm, defer,
-                                      container_name)
+                    A = Job.wrapJobFn(_testDockerCleanFn, data_dir, detached, rm, deferParam, container_name)
                     try:
                         Job.Runner.startToil(A, options)
                     except FailedJobsException:
                         # The file created by spooky_container would remain in the directory, and since
                         # it was created inside the container, it would have had uid and gid == 0 (root)
-                        # upon creation. If the defer mechanism worked, it should now be non-zero and we
-                        # check for that.
+                        # which may cause problems when docker attempts to clean up the jobstore.
                         file_stats = os.stat(test_file)
                         assert file_stats.st_gid != 0
                         assert file_stats.st_uid != 0
-                        if (rm and defer != FORGO) or defer == RM:
+
+                        if (rm and (deferParam != FORGO)) or deferParam == RM:
                             # These containers should not exist
-                            assert _containerIsRunning(container_name) is None, \
-                                'Container was not removed.'
-                        elif defer == STOP:
+                            assert _containerIsRunning(container_name) is None, 'Container was not removed.'
+
+                        elif deferParam == STOP:
                             # These containers should exist but be non-running
-                            assert _containerIsRunning(container_name) == False, \
-                                'Container was not stopped.'
+                            assert _containerIsRunning(container_name) == False, 'Container was not stopped.'
+
                         else:
                             # These containers will be running
-                            assert _containerIsRunning(container_name) == True, \
-                                'Container was not running.'
-                    finally:
-                        # Prepare for the next test.
-                        _dockerKill(container_name, RM)
```

Comment:
Why don't we need to kill this container anymore?

Reply:
There is a deferred kill on every dockerCall.

Reply:
Updated to include this again.